### DISCUSSION TITLE
Artillery script throwing 401 error for wss websocket request with authorization header token

### DISCUSSION THREAD

**Discussion by Vikrant87-lego** [artilleryio/artillery#1159] [Artillery]
Hi ,

I am facing issue where the Artillery request is throwing 401 error for websocket connect request sent with authorization headers:

config:
  target: "wss://baseurl/websocket/"
  phases:
    - duration: 60
      arrivalRate: 25
  ws:
    headers:
      authorization: "Bearer TOKEN"

scenarios:
  - name: "Connecting to Websocket"
    engine: ws
    flow:
      - connect:
          function:

Dont know what is wrong with this script. can anyone please helo me understand ?

BR,
Vikrant

**Comment by hassy** [comment]
hi @Vikrant87-lego 👋 the script itself looks OK - can you confirm that the token is valid?

**Reply by Vikrant87-lego** [reply]
Hi @hassy ,

Yes, i am using the valid bearer token but still getting this error.

**Comment by Vikrant87-lego** [comment]
I got it resolved, i was using 'BaseURL/Websocket/ as target but updated  as below:

target:'BaseURL'

and 401 error was gone

**Reply by Vikrant87-lego** [reply]
I am not getting 401 error now but getting "  Unexpected server response: 200:" instead 

can anyone please help here?
